clients/filesystem-hadoop3/src/test/java/com/datastrato/gravitino/filesystem/hadoop3/TestGravitinoVirtualFileSystem.java

```diff
@@ -187,15 +189,25 @@ public void testInternalCache() throws IOException, InterruptedException {
           localPath2.toString());
       FileSystemTestUtils.mkdirs(filesetPath2, fs);
 
-      Thread.sleep(5000);
-      assertNull(
-          ((GravitinoVirtualFileSystem) fs)
-              .getFilesetCache()
-              .getIfPresent(NameIdentifier.of(metalakeName, catalogName, schemaName, "fileset1")));
+      Awaitility.await()
+          .atMost(5, TimeUnit.SECONDS)
+          .untilAsserted(
+              () ->
+                  assertNull(
+                      ((GravitinoVirtualFileSystem) fs)
+                          .getFilesetCache()
+                          .getIfPresent(
+                              NameIdentifier.of(
+                                  metalakeName, catalogName, schemaName, "fileset1"))));
 
       // expired by time
-      Thread.sleep(5000);
-      assertEquals(0, ((GravitinoVirtualFileSystem) fs).getFilesetCache().asMap().size());
+      Awaitility.await()
+          .atMost(5, TimeUnit.SECONDS)
+          .untilAsserted(
+              () ->
+                  assertEquals(
+                      0, ((GravitinoVirtualFileSystem) fs).getFilesetCache().asMap().size()));
+
```

Comment:
Can you please add the `pollInterval()` to optimize the checking mechanism to not have to wait for 5 seconds.

Reply:
Done